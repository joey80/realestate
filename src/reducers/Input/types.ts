import { INPUT } from 'src/actions/constants';

interface Action {
  errorName: keyof SearchLocation;
  stateName: keyof SearchLocation;
  stateValue: string;
  type: keyof typeof INPUT;
}

interface Errors {
  city: boolean;
  state: boolean;
  streetAddress: boolean;
  zipCode: boolean;
}

interface SearchLocation {
  city: string;
  state: string;
  streetAddress: string;
  zipCode: string;
}

interface InitialState {
  errors: Errors;
  searchLocation: SearchLocation;
}

export type { Action, InitialState, SearchLocation };
