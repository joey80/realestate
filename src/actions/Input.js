//import { ADD_ARTICLE } from '../constants/action-types';

export const toggleInputError = payload => {
    return { type: 'TOGGLE_INPUT_ERROR', payload };
};