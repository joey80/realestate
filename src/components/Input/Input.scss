.input {

    &__field {
        display: block;
        width: 100%;
    }
}