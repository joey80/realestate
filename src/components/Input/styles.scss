input {
    background-color: #f7f7f7 !important;
    border: 1px solid rgba(172, 170, 170, 0.4);
    filter: none !important;
    margin-top: 1.5rem;
    padding: 1rem;

    &:focus {
        border: 1px solid #3624696e;
    }
}

.input {

    &__field {
        display: block;
        width: 100%;

        &--error {
            border: 1px solid #ffa8a8 !important;
            color: #c52424;
        }
    }
}