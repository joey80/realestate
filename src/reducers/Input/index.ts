import { InputReducer } from './types';

const initialState = {
  errors: {
    city: false,
    state: false,
    streetAddress: false,
    zipCode: false,
  },
  searchLocation: {
    city: '',
    state: 'Select A State',
    streetAddress: '',
    zipCode: '',
  },
};

const Input: InputReducer = (state = initialState, action) => {
  const { errorName, stateName, stateValue, type } = action;

  switch (type) {
    case 'SHOW_INPUT_ERROR': {
      return {
        ...state,
        errors: {
          ...state.errors,
          [errorName]: (state.errors[errorName] = true),
        },
      };
    }

    case 'HIDE_INPUT_ERROR': {
      return {
        ...state,
        errors: {
          ...state.errors,
          [errorName]: (state.errors[errorName] = false),
        },
      };
    }

    case 'SAVE_VALUE': {
      return {
        ...state,
        searchLocation: {
          ...state.searchLocation,
          [stateName]: (state.searchLocation[stateName] = stateValue),
        },
      };
    }

    case 'CLEAR_SEARCH_VALUES': {
      return {
        ...state,
        searchLocation: {
          ...state.searchLocation,
          streetAddress: '',
          city: '',
          state: 'Select A State',
          zipCode: '',
        },
      };
    }

    default: {
      return state;
    }
  }
};

export default Input;
