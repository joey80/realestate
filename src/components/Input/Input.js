import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import InputError from '../InputError/InputError';
import { checkValidZip } from '../Form/Helper';
import './Input.scss';

const mapStateToProps = state => ({
    errors: state.Input.errors,
    searchLocation: state.Input.searchLocation
});

const mapDispatchToProps = dispatch => ({
    showError: name => dispatch({ type: 'SHOW_INPUT_ERROR', errorName: name }),
    hideError: name => dispatch({ type: 'HIDE_INPUT_ERROR', errorName: name }),
    saveValue: (name, value) => dispatch({ type: 'SAVE_VALUE', stateName: name, stateValue: value })
});

export const Input = props => {

    const checkValue = target => {
      
        // Check for blank values
        if (target.value === '' ? props.showError(props.label) : props.hideError(props.label));
      
        // Validate zip code
        if (props.label === 'zipCode' && target.value !== '') {
          const valid = checkValidZip(target.value);
          if (valid === false ? props.showError(props.label) : props.hideError(props.label));
        } 
    };
    
    return (
        <div className="form__group">
            <label htmlFor={ props.label }>{ props.name }</label>
            <input
                type={ props.type }
                onChange={ e => { props.saveValue(props.label, e.target.value) }}
                onBlur={ e => { checkValue(e.target) }}
                className="form__field"
                name={ props.label }
            />
            <InputError
                error={ props.errors[props.label] }
                errorMessage={ props.errorMessage }
            />
        </div>
    );
};

Input.propTypes = {
    showError: PropTypes.func,
    label: PropTypes.string,
    name: PropTypes.string,
    type: PropTypes.string,
    hideError: PropTypes.func,
    saveValue: PropTypes.func,
    errors: PropTypes.object,
    errorMessage: PropTypes.string
};

export default connect(mapStateToProps, mapDispatchToProps)(Input);