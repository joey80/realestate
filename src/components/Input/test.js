import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Input from './Input';

configure({ adapter: new Adapter() });

describe('<Input />', () => {
  it('renders correctly', () => {
    shallow(<Input />);
  });
});