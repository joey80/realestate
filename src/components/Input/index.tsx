import React from 'react';
import './styles.scss';

interface InputType extends React.InputHTMLAttributes<HTMLInputElement> {
  error: boolean;
}

const Input = ({ error, ...rest }: InputType) => (
  <input {...rest} className={`input__field ${error ? 'input__field--error' : ''}`} type='text' />
);

export default Input;
