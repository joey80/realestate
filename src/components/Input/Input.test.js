import { Input } from './Input'; 

describe('Input Fields', () => {

    it('Should show error message if field is empty', () => {
        
    });

    it('Should hide error message if field is not empty', () => {

    });

    it('Should only let valid zipcodes pass', () => {

    });
});