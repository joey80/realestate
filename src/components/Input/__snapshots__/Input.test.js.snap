// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render correctly 1`] = `
<div
  className="form__group"
>
  <label
    htmlFor="streetAddress"
  />
  <input
    className="form__field"
    name="streetAddress"
    onBlur={[Function]}
    onChange={[Function]}
  />
  <InputError
    error={false}
  />
</div>
`;
